### PR TITLE
[SPARK-37852][PYTHON][INFRA] Enable flake's E741 rule in PySpark

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR enables flake's [E741](https://www.flake8rules.com/rules/E741.html) rule in PySpark codebase to comply PEP 8 (https://www.python.org/dev/peps/pep-0008/#names-to-avoid)

### Why are the changes needed?

To comply PEP 8.

### Does this PR introduce _any_ user-facing change?

No, dev-only.

### How was this patch tested?

Existing test cases should cover.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @ueshin @itholic @zero323 @xinrong-databricks FYI

File: python/pyspark/pandas/namespace.py [hunk 1]
```diff
@@ -3399,8 +3399,8 @@ def resolve(internal: InternalFrame, side: str) -> InternalFrame:
         left_join_on_columns = [scol_for(left_table, label) for label in left_join_on_names]
         right_join_on_columns = [scol_for(right_table, label) for label in right_join_on_names]
         on = reduce(
-            lambda l, r: l & r,
-            [l == r for l, r in zip(left_join_on_columns, right_join_on_columns)],
+            lambda lft, r: lft & r,
+            [lft == r for lft, r in zip(left_join_on_columns, right_join_on_columns)],
```
Comment (zero323) [inline python/pyspark/pandas/namespace.py:3403]: We could convert the other argument to something like `rgt`.

Current naming kind of suggests there is some asymmetry here and `r ` is different type of object from `lft`.

File: python/pyspark/tests/test_rdd.py [hunk 2]
```diff
@@ -615,11 +615,11 @@ def test_repartitionAndSortWithinPartitions_desc(self):
     def test_repartition_no_skewed(self):
         num_partitions = 20
         a = self.sc.parallelize(range(int(1000)), 2)
-        l = a.repartition(num_partitions).glom().map(len).collect()
-        zeros = len([x for x in l if x == 0])
+        lst = a.repartition(num_partitions).glom().map(len).collect()
+        zeros = len([x for x in lst if x == 0])
```
Comment (zero323) [inline python/pyspark/tests/test_rdd.py:619]: We iterate with `x`, we could just name it `xs`, but I am not insisting on that :)

File: python/pyspark/mllib/linalg/__init__.py [hunk 3]
```diff
@@ -128,8 +128,8 @@ def _format_float(f, digits=4):
     return s
 
 
-def _format_float_list(l):
-    return [_format_float(x) for x in l]
+def _format_float_list(lst):
```
Comment (zero323) [inline python/pyspark/mllib/linalg/__init__.py:131]: `xs`?

File: python/pyspark/ml/linalg/__init__.py [hunk 4]
```diff
@@ -125,8 +125,8 @@ def _format_float(f, digits=4):
     return s
 
 
-def _format_float_list(l):
-    return [_format_float(x) for x in l]
+def _format_float_list(lst):
+    return [_format_float(x) for x in lst]
```
Comment (zero323) [inline python/pyspark/ml/linalg/__init__.py:129]: `xs`?

**Comment by zero323** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
Thanks @zero323 for your review.

**Review by ueshin — APPROVED** [review]
LGTM.

**Pull request closed without merging**